packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx

```diff
@@ -0,0 +1,318 @@
+import { telemetryEventTypes, useTelemetry } from "@calcom/lib/telemetry";
+import { trpc } from "@calcom/trpc/react";
+import {
+  Button,
+  Form,
+  TextField,
+  Alert,
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  Label,
+  Input,
+} from "@calcom/ui";
+import { ArrowRight, Info } from "@calcom/ui/components/icon";
+
+function extractDomainFromEmail(email: string) {
+  let out = "";
+  try {
+    const match = email.match(/^(?:.*?:\/\/)?.*?(?<root>[\w\-]*(?:\.\w{2,}|\.\w{2,}\.\w{2}))(?:[\/?#:]|$)/);
+    out = (match && match.groups?.root) ?? "";
+  } catch (ignore) {}
+  return out.split(".")[0];
+}
+
+export const VerifyCodeDialog = ({
+  isOpenDialog,
+  setIsOpenDialog,
+  email,
+  onSuccess,
+}: {
+  isOpenDialog: boolean;
+  setIsOpenDialog: Dispatch<SetStateAction<boolean>>;
+  email: string;
+  onSuccess: (isVerified: boolean) => void;
+}) => {
+  const { t } = useLocale();
+  // Not using the mutation isLoading flag because after verifying we submit the underlying org creation form
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [value, onChange] = useState("");
+
+  const digits = useDigitInput({
+    acceptedCharacters: /^[0-9]$/,
+    length: 6,
+    value,
+    onChange,
+  });
+
+  const verifyCodeMutation = trpc.viewer.organizations.verifyCode.useMutation({
+    onSuccess: (data) => {
+      setIsLoading(false);
+      onSuccess(data);
+    },
+    onError: (err) => {
+      setIsLoading(false);
+      if (err.message === "invalid_code") {
+        setError(t("code_provided_invalid"));
+      }
+    },
+  });
+
+  const digitClassName = "h-12 w-12 !text-xl text-center";
+
+  return (
+    <Dialog
+      open={isOpenDialog}
+      onOpenChange={(open) => {
+        onChange("");
+        setError("");
+        setIsOpenDialog(open);
+      }}>
+      <DialogContent className="sm:max-w-md">
+        <div className="flex flex-row">
+          <div className="w-full">
+            <DialogHeader title={t("verify_your_email")} subtitle={t("enter_digit_code", { email })} />
+            <Label htmlFor="code">{t("code")}</Label>
+            <div className="flex flex-row justify-between">
+              <Input
+                className={digitClassName}
+                name="2fa1"
+                inputMode="decimal"
+                {...digits[0]}
+                autoFocus
+                autoComplete="one-time-code"
+              />
+              <Input className={digitClassName} name="2fa2" inputMode="decimal" {...digits[1]} />
+              <Input className={digitClassName} name="2fa3" inputMode="decimal" {...digits[2]} />
+              <Input className={digitClassName} name="2fa4" inputMode="decimal" {...digits[3]} />
+              <Input className={digitClassName} name="2fa5" inputMode="decimal" {...digits[4]} />
+              <Input className={digitClassName} name="2fa6" inputMode="decimal" {...digits[5]} />
```

Comment:
NIT: Lets make this a component as its used in a few places 

Doesnt need to be picked up here - created a issue

Reply:
Good catch!